src/drivers/abstract/logger.js

```diff
@@ -25,7 +26,15 @@ class Logger {
   }
 
   logQuery(sql, duration, opts) {
-    debug('[query] [%s] %s', duration, sql);
+    if (this._opts.logQuery) {
+      try {
+        this._opts.logQuery(sql, duration, opts);
+      } catch (error) {
+        console.error(error);
+      }
```

Comment:
The original implementation was intended to eliminate property checks as much as possible. Perhaps the new `logQuery` can be renamed as something like `tryLogQuery`?

Reply:
or just change the original function as the default `this.options.logQuery()`